src/Controls/src/Core/Platform/ModalNavigationManager/ModalNavigationManager.Windows.cs

```diff
@@ -40,33 +64,33 @@ void RemovePage(Page page)
 
 			var windowManager = mauiContext.GetNavigationRootManager();
 			Container.RemovePage(windowManager.RootView);
+
+			if (popping)
+			{
+				page
+					.FindMauiContext()
+					?.GetNavigationRootManager()
+					?.Disconnect();
+
+				page.Handler?.DisconnectHandler();
```

Comment:
Funny indentation, is that intended? You can divert the if to have a little nicer looking alignment if you want.